src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
+      throw new UnauthorizedException();
```

Comment:
PullProofAI

There is a potential security vulnerability in the `validate` method of the `JwtStrategy` class. The code retrieves a JWT token from Redis based on the user ID without any validation or verification, which could lead to unauthorized access if an attacker manipulates the user ID.

It's crucial to add additional checks to ensure the integrity and authenticity of the JWT token before trusting it for authentication purposes.

```diff
19,23c19,27
<     const { userId } = payload;
<     const client = this.redisService.getClient();
<     const jwtToken = await client.get(`jwt:${userId}`);
<     if (!jwtToken) {
<       throw new UnauthorizedException();


<!-- This is an auto-generated comment by PullProofAI -->